### PR TITLE
Unsupport py34; Support newer arrow and requests-oauthlib dependencies

### PR DESCRIPTION
**Pull request by jlapenna**
This also required updating calls to fetch_token to include_client_id, as needed by the withings/nokia API.

### CODE REVIEW

**Comment by brad** [comment]
@jlapenna Looks like we should drop support for Python 3.4. Can you make the change?

**Comment by jlapenna** [comment]
I can definitely do that. I've never used worked with travis CI before so it might take me a minute or six to figure out how to stop testing/supporting py3.4

**Comment by jlapenna** [comment]
Alright (I'm also new to github), I pushed my changes to my fork, which according to the comments here should maybe trigger test re-runs.

**Comment by jlapenna** [comment]
Hm. This seems to still use the tox.ini in your branch, so I'm not sure how to trigger it with this pull request.

**Comment by brad** [comment]
@jlapenna That is right, it just needs to be removed from `.travis.yml` as well

**Comment by jlapenna** [comment]
Looks correct, now. yay. I also updated the title of the pull request.

File: requirements/base.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-arrow>=0.12,<0.13
+arrow>=0.14,<0.15
```
Comment (brad) [inline requirements/base.txt]: Can we leave the lower end at 0.12 or does this change not work on < 0.14?

File: requirements/base.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 cherrypy>=17.3,<17.4
 requests>=2.20.0
 requests-oauth>=0.4.1,<0.5
-requests-oauthlib>=1.0,<1.1
+requests-oauthlib>=1.2,<1.3
```
Comment (brad) [inline requirements/base.txt]: Can we leave the lower end at 1.0 or does this change break things with < 1.2?

Comment (jlapenna) [reply requirements/base.txt]: I believe we can't use 1.1 as 1.2 introduced a breaking change to only support oauthlib 3.0 https://github.com/requests/requests-oauthlib/commit/1495328dea94b38e3675476cb889e2e3a038355b#diff-d9ae695c8119e58c2ec435b6deae2d53

File: requirements/base.txt (hunk 1 above)
Comment (jlapenna) [reply requirements/base.txt]: I haven't specifically tried it with .12, I was following the pattern elsewhere in the file where it seemed we tried to pin to approximately 1 minor version. I'll give it a shot and report back (here, and below)

**Comment by jlapenna** [comment]
I haven't done a code review in a git pull request before, so I don't know if I resolved your comments correctly. Let me know.

**Comment by jlapenna** [comment]
@brad I have made all the requested changes.

**Comment by brad** [comment]
:shipit:

**Pull request merged**